### PR TITLE
kubo: new, 0.18.1

### PR DESCRIPTION
**Pull request by dynos01**
Topic Description
-----------------
kubo: new, 0.18.1

Package(s) Affected
-------------------
kubo: new, 0.18.1

Security Update?
----------------
No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: extra-network/kubo/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,9 @@
+abinfo "Building Kubo..."
+make build
+cmd/ipfs/ipfs commands completion bash > "$SRCDIR"/ipfs-completion.bash
+
+abinfo "Installing Kubo binary..."
+install -Dvm755 "$SRCDIR"/cmd/ipfs/ipfs -t "$PKGDIR"/usr/bin/
+
+abinfo "Installing Kubo Bash completion..."
+install -Dvm644 "$SRCDIR"/ipfs-completion.bash -t "$PKGDIR"/usr/share/bash-completion/completions/
```
Comment (MingcongBai) [inline extra-network/kubo/autobuild/build]: Please ensure that a `newline` exists for end-of-file. This applies for all the files you have committed.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-network/kubo/autobuild/build (hunk 1 above)
Comment (dynos01) [reply extra-network/kubo/autobuild/build]: I've added newlines to all files commited.

**Review by MingcongBai — APPROVED** [review]

File: extra-network/kubo/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,9 @@
+abinfo "Building Kubo..."
+make build
+cmd/ipfs/ipfs commands completion bash > "$SRCDIR"/ipfs-completion.bash
```
Comment (MingcongBai) [inline extra-network/kubo/autobuild/build]: Absolute path here. `"$SRCDIR"/cmd/ipfs/ipfs ...`, it is also recommended to wrap wide lines.

File: extra-network/kubo/autobuild/prepare [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export CGO_LDFLAGS="$LDFLAGS"
+export CGO_CFLAGS="$CFLAGS"
+export CGO_CPPFLAGS="$CPPFLAGS"
+export CGO_CXXFLAGS="$CXXFLAGS"
+export GOFLAGS="-buildmode=pie -trimpath -modcacherw"
```
Comment (MingcongBai) [inline extra-network/kubo/autobuild/prepare]: All of these are not necessary. Autobuild3 defines all these flags internally.

File: extra-network/kubo/spec [hunk 4]
```diff
@@ -0,0 +1,3 @@
+VER=0.18.1
+SRCS="tbl::https://github.com/ipfs/kubo/archive/refs/tags/v$VER.zip"
+CHKSUMS="sha256::4dc65c1177fe2ff339c64063643276112efa029117d0810f14422208dd6a3c32"
```
Comment (MingcongBai) [inline extra-network/kubo/spec:3]: You should add a CHKUPDATE field indicating a way to automatically check for updates. This is done with [aosc-findupdate](https://github.com/AOSC-Dev/aosc-findupdate). We usually use the `anitya` type, using Fedora's [Anitya](https://release-monitoring.org/).

Please register an account there and submit `kubo` as one of the software to track. After you are done, you will get an update-check ID. For instance, Bash, at https://release-monitoring.org/project/166/, means that it has an ID of 166.

In this case, then, the CHKUPDATE line should go something like this.

```
CHKUPDATE="anitya::id=166"
```

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-network/kubo/spec (hunk 4 above)
Comment (MingcongBai) [reply extra-network/kubo/spec:3]: We require this information for all packages, where possible.

**Comment by dynos01** [comment]
kubo

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**